app/components/user-status-modal.js

```diff
@@ -0,0 +1,104 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { toastNotificationTimeoutOptions } from '../constants/toast-notification';
+import {
+  warningMessageForIdle,
+  warningMessageForOOO,
+  warningMessageForFromField,
+  warningMessageForUntilField,
+  userStates,
+} from '../constants/user-status';
+
+export default class FormStatusModal extends Component {
+  @service toast;
+  @tracked currentStatus;
+  @tracked newStatus;
+  @tracked fromDate = '';
+  @tracked untilDate = '';
+  @tracked reason = '';
+
+  @action
+  updateValue(event) {
+    const { name, value } = event.target;
+    if (name === 'fromDate') {
+      this.fromDate = value;
+    } else if (name === 'untilDate') {
+      this.untilDate = value;
+    } else if (name === 'reason') {
+      this.reason = value;
+    }
+  }
+
+  @action
+  async getCurrentStatusObj() {
+    let from;
+    let until;
+    if (!this.fromDate) {
+      this.toast.error(
+        warningMessageForFromField,
+        '',
+        toastNotificationTimeoutOptions
+      );
+      return;
+    }
```

Comment:
Can we please have a line space after each if block?

Reply:
yeah, that will make it more readable. Done